vulture/config.py

```diff
@@ -0,0 +1,168 @@
+    )
+    parser.add_argument(
+        "--exclude",
+        metavar="PATTERNS",
+        type=csv,
+        help="Comma-separated list of paths to ignore (e.g.,"
+        ' "*settings.py,docs/*.py"). {glob_help} A PATTERN without glob'
+        " wildcards is treated as *PATTERN*.".format(**locals()),
+    )
+    parser.add_argument(
+        "--ignore-decorators",
+        metavar="PATTERNS",
+        type=csv,
+        help="Comma-separated list of decorators. Functions and classes using"
+        ' these decorators are ignored (e.g., "@app.route,@require_*").'
+        " {glob_help}".format(**locals()),
+    )
+    parser.add_argument(
+        "--ignore-names",
+        metavar="PATTERNS",
+        type=csv,
+        default=None,
+        help='Comma-separated list of names to ignore (e.g., "visit_*,do_*").'
+        " {glob_help}".format(**locals()),
+    )
+    parser.add_argument(
+        "--make-whitelist",
+        action="store_true",
+        help="Report unused code in a format that can be added to a"
+        " whitelist module.",
+    )
+    parser.add_argument(
+        "--min-confidence",
+        type=int,
+        default=MIN_CONFIDENCE_DEFAULT,
+        help="Minimum confidence (between 0 and 100) for code to be"
+        " reported as unused.",
+    )
+    parser.add_argument(
+        "--sort-by-size",
+        action="store_true",
+        help="Sort unused functions and classes by their lines of code.",
+    )
+    parser.add_argument("-v", "--verbose", action="store_true")
+    parser.add_argument("--version", action="version", version=version)
+    namespace = parser.parse_args(args)
+    output = Config(
+        paths=namespace.paths,
+        exclude=namespace.exclude,
+        ignore_decorators=namespace.ignore_decorators,
+        ignore_names=namespace.ignore_names,
+        make_whitelist=namespace.make_whitelist,
+        min_confidence=namespace.min_confidence,
+        sort_by_size=namespace.sort_by_size,
+        verbose=namespace.verbose,
+    )
+    return output
+
+
+class Config(Dict[str, Any]):
+    """
+    A config object wrapping various variables and used to abstract away the
+    difference between CLI-arg parsing and TOML loading.
+    """
+
+    # pylint: disable=too-many-arguments, too-few-public-methods
+    # pylint: disable=useless-object-inheritance
+    # pylint: disable=too-many-instance-attributes
+
+    def __getattribute__(self, name):
+        # type: (str) -> Any
+        if name in self:
+            return self[name]
+        return super(Config, self).__getattribute__(name)
+
+
+def make_config():
+    # type: () -> Config
+    """
+    Returns a config object for vulture, merging both ``pyproject.toml`` and
+    CLI arguments (CLI will have precedence).
+    """
+    if exists("pyproject.toml"):
+        with open("pyproject.toml") as toml_config:
+            app_config = _parse_toml(toml_config)
+    cli_config = _parse_args()
+    app_config.update(cli_config)
```

Comment:
I think the CLI defaults will override all TOML settings, won't they? In any case, we should add a test for this.

Reply:
Without this line, the CLI defaults will not automatically override them (unless I'm missing something). I agree about the test, although I need to refactor some things for simple DI (because of the `open()` call).

Reply:
Addressed in ae44f09